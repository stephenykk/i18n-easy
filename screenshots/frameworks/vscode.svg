<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="17pt" height="16pt" viewBox="0 0 17 16" version="1.1">
<g id="surface1">
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(14.117647%,53.72549%,79.215686%);fill-opacity:1;" d="M 0.492188 5.613281 C 0.492188 5.613281 0.0820312 5.347656 0.570312 4.976562 L 1.703125 4.035156 C 1.703125 4.035156 2.03125 3.714844 2.375 3.988281 L 12.839844 11.390625 L 12.839844 14.9375 C 12.839844 14.9375 12.832031 15.496094 12.070312 15.433594 Z M 0.492188 5.613281 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(6.27451%,43.921569%,70.196078%);fill-opacity:1;" d="M 3.1875 7.90625 L 0.492188 10.195312 C 0.492188 10.195312 0.210938 10.386719 0.492188 10.730469 L 1.742188 11.796875 C 1.742188 11.796875 2.042969 12.09375 2.480469 11.753906 L 5.339844 9.730469 Z M 3.1875 7.90625 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(3.137255%,46.666667%,72.54902%);fill-opacity:1;" d="M 7.925781 7.925781 L 12.871094 4.398438 L 12.839844 0.863281 C 12.839844 0.863281 12.628906 0.09375 11.925781 0.496094 L 5.339844 6.09375 Z M 7.925781 7.925781 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(23.529412%,60%,83.137255%);fill-opacity:1;" d="M 12.070312 15.441406 C 12.355469 15.714844 12.699219 15.625 12.699219 15.625 L 16.558594 13.851562 C 17.054688 13.539062 16.984375 13.148438 16.984375 13.148438 L 16.984375 2.550781 C 16.984375 2.085938 16.472656 1.925781 16.472656 1.925781 L 13.132812 0.425781 C 12.402344 0 11.925781 0.496094 11.925781 0.496094 C 11.925781 0.496094 12.539062 0.0820312 12.839844 0.863281 L 12.839844 14.878906 C 12.839844 14.972656 12.816406 15.066406 12.773438 15.152344 C 12.683594 15.3125 12.496094 15.46875 12.035156 15.40625 Z M 12.070312 15.441406 "/>
</g>
</svg>
