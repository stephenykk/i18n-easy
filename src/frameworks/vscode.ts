import { basename } from 'path'
import { TextDocument } from 'vscode'
import { Framework } from './base'
import { LanguageId } from '~/utils'

class VSCodeFramework extends Framework {
  id = 'vscode'
  display = 'VS Code'

  detection = {
    packageJSON: [
      'vscode',
      '@types/vscode',
    ],
  }

  enabledParsers = ['json']

  languageIds: LanguageId[] = [
    'json',
    'javascript',
    'typescript',
  ]

  pathMatcher = () => 'package.nls.?{locale?}.json'

  usageMatchRegex = (languageIds?: string, filename?: string) => {
    if (filename && basename(filename) === 'package.json')
      return '"%({key})%"'
    // for visualize the regex, you can use https://regexper.com/
    return '(?:i18n[ (]path=|v-t=[\'"`{]|(?:this\\.|\\$|i18n\\.)(?:(?:d|n)\\(.*?,|(?:t|tc|te)\\()\\s*)[\'"`]({key})[\'"`]'
  }

  refactorTemplates(keypath: string, args?: string[], doc?: TextDocument) {
    if (doc?.languageId === 'json') {
      return [
        `%${keypath}%`,
        keypath,
      ]
    }

    return [
      `i18n.t('${keypath}')`,
      keypath,
    ]
  }
}

export default VSCodeFramework
