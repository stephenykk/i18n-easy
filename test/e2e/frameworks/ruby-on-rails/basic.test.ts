import { window } from 'vscode'
import { getExt, openFile, Global, is, not, setupTest, timeout, KeyDetector, expect } from '../../ctx'

setupTest('Ruby on Rails', () => {
  it('opens entry file', async() => {
    await openFile('Gemfile')
  })

  it('is active', () => {
    const ext = getExt()
    is(ext?.isActive, true)
  })

  it('enables correct frameworks', async() => {
    not(Global, undefined)
    is(Global.enabled, true)
    is(Global.enabledFrameworks.length, 1)
    is(Global.enabledFrameworks[0].id, 'ruby-rails')
  })

  it('get keys', async() => {
    await openFile('app/views/pages/index.html.erb')
    await timeout(500)
    const keys = KeyDetector.getKeys(window.activeTextEditor!.document)
    expect(keys).to.matchSnapshot()
  })
})
