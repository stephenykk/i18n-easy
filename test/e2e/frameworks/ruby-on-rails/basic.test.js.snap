// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ruby on Rails get keys 1`] = `
Array [
  Object {
    "end": 19,
    "key": "pages.index.welcome",
    "quoted": true,
    "start": 11,
  },
  Object {
    "end": 53,
    "key": "pages.index.bold_text.html",
    "quoted": true,
    "start": 38,
  },
  Object {
    "end": 73,
    "key": "welcome",
    "quoted": true,
    "start": 66,
  },
  Object {
    "end": 269,
    "key": "pages.index.new_post",
    "quoted": true,
    "start": 260,
  },
  Object {
    "end": 315,
    "key": "pages.index.posts",
    "quoted": true,
    "start": 309,
  },
  Object {
    "end": 368,
    "key": "pages.index.new_post",
    "quoted": false,
    "start": 348,
  },
]
`;
